crates/sp-domains/src/fraud_proof.rs

```diff
@@ -70,6 +70,11 @@ impl ExecutionPhase {
     }
 }
 
+/// Trait to derive extrinsics.
```

Comment:
Comments everywhere say generic "derive extrinsics", but commit message says "derive timestamp extrinsic". I'm a bit confused, is this for all extrinsics, just one extrinsic, how does it do exactly? The fact that you introduce things that are underdocumented, but not used, makes it harder to review because I have no idea why it is helpful and where, it is just dead code.